contents/tutorials/nextjs-surveys.md

```diff
@@ -0,0 +1,536 @@
+2. `"survey dismissed"`
+3. `"survey sent"` (for responses)
+
+You can capture these events using `posthog.capture()`:
+
+```react
+  // ... rest of your code ...
+
+  const handleDismiss = () => {
+    setShowSurvey(false);
+    console.log("Survey dismissed!");
+    localStorage.setItem(`hasInteractedWithSurvey_${surveyID}`, 'true');
+    posthog.capture("survey dismissed", {
+      $survey_id: surveyID // required
+    })
+  };
+
+  const handleSubmit = (value) => {
+    setShowSurvey(false);
+    console.log("User submitted:", value);
+    localStorage.setItem(`hasInteractedWithSurvey_${surveyID}`, 'true');  
+    posthog.capture("survey sent", {
+      $survey_id: surveyID, // required
+      $survey_response: value // required
+    })
+  };
+
+  useEffect(() => {
+    if (posthog && surveyID && showSurvey) {
+      posthog.capture("survey seen", {
+        $survey_id: surveyID // required
+      })
+    }
+  }, [showSurvey, surveyID, posthog])
+
+ // ... rest of your code ...
+```
+
+Altogether, your code should look like this:
+
+```react
+// app/page.js
+'use client'
+
+import { useState, useEffect } from 'react'
+import styles from './page.module.css'
+import Survey from './Survey';
+import { usePostHog } from 'posthog-js/react';
+
+export default function Home() {
+  const [showSurvey, setShowSurvey] = useState(true);
+  const [surveyTitle, setSurveyTitle] = useState(false);
+  const [surveyID, setSurveyID] = useState(false);
+  const posthog = usePostHog()
+
+  useEffect(() => {
+    posthog.getActiveMatchingSurveys((surveys) => {
+      if (surveys.length > 0) {
+        const survey = surveys[0];
+        setSurveyID(survey.id);
+        setSurveyTitle(survey.questions[0].question)
+      }
+    }); 
+  }, [posthog])
+
+  const handleDismiss = () => {
+    setShowSurvey(false);
+    console.log("Survey dismissed!");
+    localStorage.setItem(`hasInteractedWithSurvey_${surveyID}`, 'true');
+    posthog.capture("survey dismissed", {
+      $survey_id: surveyID // required
+    })
+  };
+
+  const handleSubmit = (value) => {
+    setShowSurvey(false);
+    console.log("User submitted:", value);
+    localStorage.setItem(`hasInteractedWithSurvey_${surveyID}`, 'true');  
+    posthog.capture("survey sent", {
+      $survey_id: surveyID, // required
+      $survey_response: value // required
+    })
+  };
+
+  useEffect(() => {
+    if (posthog && surveyID && showSurvey) {
+      posthog.capture("survey seen", {
+        $survey_id: surveyID // required
+      })
+    }
+  }, [showSurvey, surveyID, posthog])
+
+  return (
+    <main className={styles.main}>
+      <div className="App">
+        <h1>This is our Next.js survey tutorial</h1>
+        {showSurvey && (
+        <Survey
+          title={surveyTitle}
+          onDismiss={handleDismiss}
+          onSubmit={handleSubmit}
+        />)}
+      </div>
+    </main>
+  )
+}
+```
+
+Our survey is now ready to go! The next step is to submit responses and view your results.
```

Comment:
Note on removing flickering with bootstrapping?

Reply:
Unfortunately we'd need to call `posthog.getActiveMatchingSurveys` on the client and wait for the response, so even if we implement bootstrapping, there will be still a "flicker" (its more a delay than a flicker)